### ISSUE TITLE
CI fails on master with `ModuleNotFoundError` for MacOS

### ISSUE THREAD

**Issue by tanujkhattar** [quantumlib/Cirq#6147]
**Description of the issue**
The issue seems to be related to pandas. See logs from a failed run - https://github.com/quantumlib/Cirq/actions/runs/5328915438/jobs/9654061191?pr=6122

<details>
```
Run check/pytest -n auto --ignore=cirq-core/cirq/contrib
/Users/runner/hostedtoolcache/Python/3.7.17/x[6](https://github.com/quantumlib/Cirq/actions/runs/5328915438/jobs/9654061191?pr=6122#step:6:7)4/lib/python3.[7](https://github.com/quantumlib/Cirq/actions/runs/5328915438/jobs/9654061191?pr=6122#step:6:8)/site-packages/xdist/dsession.py:70: PytestDeprecationWarning: The hookimpl DSession.pytest_sessionstart uses old-style configuration options (marks or attributes).
============================= test session starts ==============================
Please use the pytest.hookimpl(trylast=True) decorator instead
platform darwin -- Python 3.7.17, pytest-7.3.2, pluggy-1.0.0
 to configure the hooks.
Using --randomly-seed=1267634636
 See https://docs.pytest.org/en/latest/deprecations.html#configuring-hook-specs-impls-using-markers
rootdir: /Users/runner/work/Cirq/Cirq
  @pytest.mark.trylast
configfile: pyproject.toml
/Users/runner/hostedtoolcache/Python/3.7.17/x64/lib/python3.7/site-packages/xdist/dsession.py:93: PytestDeprecationWarning: The hookimpl DSession.pytest_xdist_make_scheduler uses old-style configuration options (marks or attributes).
plugins: asyncio-0.21.0, cov-4.1.0, xdist-2.2.1, randomly-3.12.0, forked-1.6.0, anyio-3.7.0
Please use the pytest.hookimpl(trylast=True) decorator instead
asyncio: mode=strict
 to configure the hooks.
gw0 I / gw1 I / gw2 I
 See https://docs.pytest.org/en/latest/deprecations.html#configuring-hook-specs-impls-using-markers
gw0 [0] / gw1 [0] / gw2 [0]
  @pytest.mark.trylast


==================================== ERRORS ====================================
________________________ ERROR collecting test session _________________________
../../../hostedtoolcache/Python/3.7.17/x64/lib/python3.7/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1006: in _gcd_import
    ???
<frozen importlib._bootstrap>:9[8](https://github.com/quantumlib/Cirq/actions/runs/5328915438/jobs/9654061191?pr=6122#step:6:9)3: in _find_and_load
    ???
<frozen importlib._bootstrap>:[9](https://github.com/quantumlib/Cirq/actions/runs/5328915438/jobs/9654061191?pr=6122#step:6:10)53: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:219: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:[10](https://github.com/quantumlib/Cirq/actions/runs/5328915438/jobs/9654061191?pr=6122#step:6:11)06: in _gcd_import
    ???
<frozen importlib._bootstrap>:983: in _find_and_load
    ???
<frozen importlib._bootstrap>:967: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:677: in _load_unlocked
    ???
<frozen importlib._bootstrap_external>:728: in exec_module
    ???
<frozen importlib._bootstrap>:219: in _call_with_frames_removed
    ???
cirq-aqt/cirq_aqt/__init__.py:19: in <module>
    from cirq_aqt.aqt_sampler import AQTSampler, AQTSamplerLocalSimulator
cirq-aqt/cirq_aqt/aqt_sampler.py:33: in <module>
    import cirq
cirq-core/cirq/__init__.py:19: in <module>
    from cirq._compat import __cirq_debug__, with_debug
cirq-core/cirq/_compat.py:31: in <module>
    import pandas as pd
../../../hostedtoolcache/Python/3.7.17/x64/lib/python3.7/site-packages/pandas/__init__.py:50: in <module>
    from pandas.core.api import (
../../../hostedtoolcache/Python/3.7.17/x64/lib/python3.7/site-packages/pandas/core/api.py:48: in <module>
    from pandas.core.groupby import (
../../../hostedtoolcache/Python/3.7.17/x64/lib/python3.7/site-packages/pandas/core/groupby/__init__.py:1: in <module>
    from pandas.core.groupby.generic import (
../../../hostedtoolcache/Python/3.7.17/x64/lib/python3.7/site-packages/pandas/core/groupby/generic.py:73: in <module>
    from pandas.core.frame import DataFrame
../../../hostedtoolcache/Python/3.7.17/x64/lib/python3.7/site-packages/pandas/core/frame.py:[12](https://github.com/quantumlib/Cirq/actions/runs/5328915438/jobs/9654061191?pr=6122#step:6:13)9: in <module>
    from pandas.core import (
../../../hostedtoolcache/Python/3.7.[17](https://github.com/quantumlib/Cirq/actions/runs/5328915438/jobs/9654061191?pr=6122#step:6:18)/x64/lib/python3.7/site-packages/pandas/core/generic.py:122: in <module>
    from pandas.core.describe import describe_ndframe
../../../hostedtoolcache/Python/3.7.17/x64/lib/python3.7/site-packages/pandas/core/describe.py:39: in <module>
    from pandas.io.formats.format import format_percentiles
../../../hostedtoolcache/Python/3.7.17/x64/lib/python3.7/site-packages/pandas/io/formats/format.py:99: in <module>
    from pandas.io.common import stringify_path
../../../hostedtoolcache/Python/3.7.17/x64/lib/python3.7/site-packages/pandas/io/common.py:4: in <module>
    import bz2
../../../hostedtoolcache/Python/3.7.17/x64/lib/python3.7/bz2.py:[19](https://github.com/quantumlib/Cirq/actions/runs/5328915438/jobs/9654061191?pr=6122#step:6:20): in <module>
    from _bz2 import BZ2Compressor, BZ2Decompressor
E   ModuleNotFoundError: No module named '_bz2'
________________________ ERROR collecting test session _________________________
../../../hostedtoolcache/Python/3.7.17/x64/lib/python3.7/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1006: in _gcd_import
    ???
<frozen importlib._bootstrap>:983: in _find_and_load
    ???
<frozen importlib._bootstrap>:953: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:[21](https://github.com/quantumlib/Cirq/actions/runs/5328915438/jobs/9654061191?pr=6122#step:6:22)9: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:1006: in _gcd_import
    ???
<frozen importlib._bootstrap>:983: in _find_and_load
    ???
<frozen importlib._bootstrap>:967: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:677: in _load_unlocked
    ???
<frozen importlib._bootstrap_external>:728: in exec_module
    ???
<frozen importlib._bootstrap>:219: in _call_with_frames_removed
    ???
cirq-aqt/cirq_aqt/__init__.py:19: in <module>
    from cirq_aqt.aqt_sampler import AQTSampler, AQTSamplerLocalSimulator
cirq-aqt/cirq_aqt/aqt_sampler.py:33: in <module>
    import cirq
cirq-core/cirq/__init__.py:19: in <module>
    from cirq._compat import __cirq_debug__, with_debug
cirq-core/cirq/_compat.py:31: in <module>
    import pandas as pd
../../../hostedtoolcache/Python/3.7.17/x64/lib/python3.7/site-packages/pandas/__init__.py:50: in <module>
    from pandas.core.api import (
../../../hostedtoolcache/Python/3.7.17/x64/lib/python3.7/site-packages/pandas/core/api.py:48: in <module>
    from pandas.core.groupby import (
../../../hostedtoolcache/Python/3.7.17/x64/lib/python3.7/site-packages/pandas/core/groupby/__init__.py:1: in <module>
    from pandas.core.groupby.generic import (
../../../hostedtoolcache/Python/3.7.17/x64/lib/python3.7/site-packages/pandas/core/groupby/generic.py:73: in <module>
    from pandas.core.frame import DataFrame
../../../hostedtoolcache/Python/3.7.17/x64/lib/python3.7/site-packages/pandas/core/frame.py:129: in <module>
    from pandas.core import (
../../../hostedtoolcache/Python/3.7.17/x64/lib/python3.7/site-packages/pandas/core/generic.py:1[22](https://github.com/quantumlib/Cirq/actions/runs/5328915438/jobs/9654061191?pr=6122#step:6:23): in <module>
    from pandas.core.describe import describe_ndframe
../../../hostedtoolcache/Python/3.7.17/x64/lib/python3.7/site-packages/pandas/core/describe.py:39: in <module>
    from pandas.io.formats.format import format_percentiles
../../../hostedtoolcache/Python/3.7.17/x64/lib/python3.7/site-packages/pandas/io/formats/format.py:99: in <module>
    from pandas.io.common import stringify_path
../../../hostedtoolcache/Python/3.7.17/x64/lib/python3.7/site-packages/pandas/io/common.py:4: in <module>
    import bz2
../../../hostedtoolcache/Python/3.7.17/x64/lib/python3.7/bz2.py:19: in <module>
    from _bz2 import BZ2Compressor, BZ2Decompressor
E   ModuleNotFoundError: No module named '_bz2'
________________________ ERROR collecting test session _________________________
../../../hostedtoolcache/Python/3.7.17/x64/lib/python3.7/importlib/__init__.py:1[27](https://github.com/quantumlib/Cirq/actions/runs/5328915438/jobs/9654061191?pr=6122#step:6:28): in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1006: in _gcd_import
    ???
<frozen importlib._bootstrap>:983: in _find_and_load
    ???
<frozen importlib._bootstrap>:953: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:219: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:1006: in _gcd_import
    ???
<frozen importlib._bootstrap>:983: in _find_and_load
    ???
<frozen importlib._bootstrap>:967: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:677: in _load_unlocked
    ???
<frozen importlib._bootstrap_external>:7[28](https://github.com/quantumlib/Cirq/actions/runs/5328915438/jobs/9654061191?pr=6122#step:6:29): in exec_module
    ???
<frozen importlib._bootstrap>:219: in _call_with_frames_removed
    ???
cirq-aqt/cirq_aqt/__init__.py:19: in <module>
    from cirq_aqt.aqt_sampler import AQTSampler, AQTSamplerLocalSimulator
cirq-aqt/cirq_aqt/aqt_sampler.py:33: in <module>
    import cirq
cirq-core/cirq/__init__.py:19: in <module>
    from cirq._compat import __cirq_debug__, with_debug
cirq-core/cirq/_compat.py:31: in <module>
    import pandas as pd
../../../hostedtoolcache/Python/3.7.17/x64/lib/python3.7/site-packages/pandas/__init__.py:50: in <module>
    from pandas.core.api import (
../../../hostedtoolcache/Python/3.7.17/x64/lib/python3.7/site-packages/pandas/core/api.py:48: in <module>
    from pandas.core.groupby import (
../../../hostedtoolcache/Python/3.7.17/x64/lib/python3.7/site-packages/pandas/core/groupby/__init__.py:1: in <module>
    from pandas.core.groupby.generic import (
../../../hostedtoolcache/Python/3.7.17/x64/lib/python3.7/site-packages/pandas/core/groupby/generic.py:73: in <module>
    from pandas.core.frame import DataFrame
../../../hostedtoolcache/Python/3.7.17/x64/lib/python3.7/site-packages/pandas/core/frame.py:1[29](https://github.com/quantumlib/Cirq/actions/runs/5328915438/jobs/9654061191?pr=6122#step:6:30): in <module>
    from pandas.core import (
../../../hostedtoolcache/Python/3.7.17/x64/lib/python3.7/site-packages/pandas/core/generic.py:122: in <module>
    from pandas.core.describe import describe_ndframe
../../../hostedtoolcache/Python/3.7.17/x64/lib/python3.7/site-packages/pandas/core/describe.py:[39](https://github.com/quantumlib/Cirq/actions/runs/5328915438/jobs/9654061191?pr=6122#step:6:40): in <module>
    from pandas.io.formats.format import format_percentiles
../../../hostedtoolcache/Python/3.7.17/x[64](https://github.com/quantumlib/Cirq/actions/runs/5328915438/jobs/9654061191?pr=6122#step:6:65)/lib/python3.7/site-packages/pandas/io/formats/format.py:99: in <module>
    from pandas.io.common import stringify_path
../../../hostedtoolcache/Python/3.7.17/x64/lib/python3.7/site-packages/pandas/io/common.py:4: in <module>
    import bz2
../../../hostedtoolcache/Python/3.7.17/x64/lib/python3.7/bz2.py:19: in <module>
    from _bz2 import BZ2Compressor, BZ2Decompressor
E   ModuleNotFoundError: No module named '_bz2'
=========================== short test summary info ============================
ERROR  - ModuleNotFoundError: No module named '_bz2'
ERROR  - ModuleNotFoundError: No module named '_bz2'
ERROR  - ModuleNotFoundError: No module named '_bz2'
============================== 3 errors in 5.[83](https://github.com/quantumlib/Cirq/actions/runs/5328915438/jobs/9654061191?pr=6122#step:6:84)s ===============================
Error: Process completed with exit code 1.
```
</details>
**Cirq version**
1.2.0.dev

**Comment by pavoljuhas** [comment]
Recent upgrade of GitHub runner images appears to include broken Python 3.7.17, where
`python -c 'import bz2'` fails with the same ModuleNotFoundError exception as above.

Reported as https://github.com/actions/runner-images/issues/7764

**Comment by pavoljuhas** [comment]
Another related bug report: https://github.com/actions/setup-python/issues/682

**Comment by pavoljuhas** [comment]
Fixed by #6151
